kubernetes/data_source_kubernetes_namespace.go

```diff
@@ -0,0 +1,73 @@
+package kubernetes
+
+import (
+	"log"
+
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	v1 "k8s.io/api/core/v1"
+	meta_v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func dataSourceKubernetesNamespace() *schema.Resource {
+	return &schema.Resource{
+		Read: dataSourceKubernetesNamespaceRead,
+
+		Schema: map[string]*schema.Schema{
+			"metadata": metadataSchema("namespace", false),
+			"spec": {
+				Type:        schema.TypeList,
+				Description: "Spec defines the behavior of the Namespace.",
+				Computed:    true,
+				MaxItems:    1,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"finalizers": {
+							Type:        schema.TypeList,
+							Description: "Finalizers is an opaque list of values that must be empty to permanently remove object from storage.",
+							Optional:    true,
+							Elem: &schema.Schema{
+								Type: schema.TypeString,
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func dataSourceKubernetesNamespaceRead(d *schema.ResourceData, meta interface{}) error {
+	name := d.Get("metadata.0.name").(string)
+	d.SetId(name)
+	conn := meta.(*KubeClientsets).MainClientset
+
+	log.Printf("[INFO] Reading namespace %s", name)
+	namespace, err := conn.CoreV1().Namespaces().Get(name, meta_v1.GetOptions{})
+	if err != nil {
+		log.Printf("[DEBUG] Received error: %#v", err)
+		return err
+	}
+	log.Printf("[INFO] Received namespace: %#v", namespace)
```

Comment:
Probably `DEBUG` level stuff.

Reply:
Tanken from the resource read. It's INFO over there too, also in every other resource. Thought I'd stay consistent.